src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -288,6 +292,31 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		}
 	}
 
+	public async changePasswordFunction(connection: IConnectionProfile, params: INewConnectionParams, uri: string, oldPassword: string, newPassword: string): Promise<void> {
+		if (oldPassword !== newPassword) {
+			this.showErrorDialog(Severity.Error, Constants.sqlPasswordChangeErrorHeader, Constants.sqlPasswordMismatchMessage);
+			return Promise.reject(new Error(Constants.sqlPasswordChangeErrorHeader));
+		}
+		let passwordChangeResult = await this._connectionManagementService.sendChangePassword(connection, uri, newPassword);
+		if (!passwordChangeResult.result) {
+			let detailMessage = passwordChangeResult.errorMessage;
+			if (detailMessage.indexOf(Constants.sqlPasswordDNMReqs) !== -1) {
```

Comment:
Why would you not just include these additional message details in the original error coming from STS? ADS shouldn't need to know anything about the specific details of the error or how to fix it.